mmrazor/models/mutables/mutable_channel/groups/mutable_channel_group.py

```diff
@@ -0,0 +1,248 @@
+                        in_channels = module.num_features
+                    elif isinstance(module, nn.Linear):
+                        in_channels = module.in_features
+                    else:
+                        raise NotImplementedError()
+                    module.register_mutable_attr('in_channels',
+                                                 container_class(in_channels))
+                if module.get_mutable_attr('out_channels') is None:
+                    out_channels = 0
+                    if isinstance(module, nn.Conv2d):
+                        out_channels = module.out_channels
+                    elif isinstance(module, nn.modules.batchnorm._BatchNorm):
+                        out_channels = module.num_features
+                    elif isinstance(module, nn.Linear):
+                        out_channels = module.out_features
+                    else:
+                        raise NotImplementedError()
+                    module.register_mutable_attr('out_channels',
+                                                 container_class(out_channels))
+
+    def _register_mutable_channel(self, mutable_channel: BaseMutableChannel):
+
+        # register mutable_channel
+        for channel in self.input_related + self.output_related:
+            module = channel.module
+            if isinstance(module, DynamicChannelMixin):
+                container: MutableChannelContainer
+                if channel.output_related and module.get_mutable_attr(
+                        'out_channels') is not None:
+                    container = module.get_mutable_attr('out_channels')
+                elif channel.output_related is False \
+                        and module.get_mutable_attr('in_channels') is not None:
+                    container = module.get_mutable_attr('in_channels')
+                else:
+                    raise NotImplementedError()
+
+                if channel.expand_ratio == 1:
+                    mutable_channel_ = mutable_channel
+                    start = channel.start
+                    end = channel.end
+                else:
+                    mutable_channel_ = mutable_channel.expand_mutable_channel(
+                        channel.expand_ratio)
+                    start = channel.start
+                    end = channel.start + (
+                        channel.end - channel.start) * channel.expand_ratio
+                if (start, end) in container.mutable_channels:
+                    # TODO refine assert
+                    existed = container.mutable_channels[(start, end)]
+
+                    assert mutable_channel is existed \
+                        or mutable_channel_ is list(
+                            existed._trace_source_mutables)[0]
+                else:
+                    container.register_mutable(mutable_channel_, start, end)
+
+
+MUTABLECHANNELGROUP = TypeVar('MUTABLECHANNELGROUP', bound=MutableChannelGroup)
```

Comment:
All uppercase is unfriendly, maybe it can be changed to `GroupType` or `ChannelGroupType`. 
Many typevars have this problem.

Reply:
I have applied MUTABLECHANNELGROUP ->  ChannelGroupType